frontend/components/customizer/query-string-builder.tsx

```diff
@@ -24,20 +27,44 @@ const QueryParamCaption = styled(BuilderCaption)`
   margin: 5px;
 `
 
+type BadgeOptionName = 'style' | 'label' | 'color' | 'logo' | 'logoColor'
+
+interface BadgeOptionInfo {
+  name: BadgeOptionName
+  label?: string
+  shieldsDefaultValue?: string
+}
+
 const supportedBadgeOptions = [
   { name: 'style', shieldsDefaultValue: 'flat' },
   { name: 'label', label: 'override label' },
   { name: 'color', label: 'override color' },
   { name: 'logo', label: 'named logo' },
   { name: 'logoColor', label: 'override logo color' },
-]
+] as BadgeOptionInfo[]
 
-function getBadgeOption(name) {
-  return supportedBadgeOptions.find(opt => opt.name === name)
+function getBadgeOption(name: BadgeOptionName): BadgeOptionInfo {
+  const result = supportedBadgeOptions.find(opt => opt.name === name)
+  if (!result) {
+    throw Error(`Unknown badge option: ${name}`)
+  }
+  return result
 }
 
-function getQueryString({ queryParams, badgeOptions }) {
-  const outQuery = {}
+function getQueryString({
+  queryParams,
+  badgeOptions,
+}: {
+  queryParams: Record<string, string | boolean>
+  badgeOptions: Record<BadgeOptionName, string | undefined>
+}): {
+  queryString: string
+  isComplete: boolean
+} {
+  // Use `string | null`, because `query-string` renders e.g.
```

Comment:
Should this comment be `string | boolean | null` to match the `Record` line below?

Reply:
Ha! Actually `boolean` should not be in there. I added that to silence a type warning but when I read the comment it's clear that the entire point is to convert the booleans to nulls.